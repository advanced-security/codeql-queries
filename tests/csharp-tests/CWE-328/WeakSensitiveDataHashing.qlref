CWE-328/WeakSensitiveDataHashing.ql