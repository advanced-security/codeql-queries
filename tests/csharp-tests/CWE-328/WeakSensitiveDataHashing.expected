edges
| WeakHashingAlgorithms.cs:52:24:52:58 | call to method GetBytes : Byte[] | WeakHashingAlgorithms.cs:53:42:53:46 | access to local variable bytes |
| WeakHashingAlgorithms.cs:52:50:52:57 | access to parameter password : String | WeakHashingAlgorithms.cs:52:24:52:58 | call to method GetBytes : Byte[] |
| WeakHashingAlgorithms.cs:65:24:65:58 | call to method GetBytes : Byte[] | WeakHashingAlgorithms.cs:66:43:66:47 | access to local variable bytes |
| WeakHashingAlgorithms.cs:65:50:65:57 | access to parameter password : String | WeakHashingAlgorithms.cs:65:24:65:58 | call to method GetBytes : Byte[] |
nodes
| WeakHashingAlgorithms.cs:52:24:52:58 | call to method GetBytes : Byte[] | semmle.label | call to method GetBytes : Byte[] |
| WeakHashingAlgorithms.cs:52:50:52:57 | access to parameter password : String | semmle.label | access to parameter password : String |
| WeakHashingAlgorithms.cs:53:42:53:46 | access to local variable bytes | semmle.label | access to local variable bytes |
| WeakHashingAlgorithms.cs:65:24:65:58 | call to method GetBytes : Byte[] | semmle.label | call to method GetBytes : Byte[] |
| WeakHashingAlgorithms.cs:65:50:65:57 | access to parameter password : String | semmle.label | access to parameter password : String |
| WeakHashingAlgorithms.cs:66:43:66:47 | access to local variable bytes | semmle.label | access to local variable bytes |
subpaths
#select
| WeakHashingAlgorithms.cs:53:42:53:46 | access to local variable bytes | WeakHashingAlgorithms.cs:52:50:52:57 | access to parameter password : String | WeakHashingAlgorithms.cs:53:42:53:46 | access to local variable bytes | Insecure hashing algorithm (MD5) depends on $@. | WeakHashingAlgorithms.cs:52:50:52:57 | access to parameter password | sensitive data (access to parameter password) |
| WeakHashingAlgorithms.cs:66:43:66:47 | access to local variable bytes | WeakHashingAlgorithms.cs:65:50:65:57 | access to parameter password : String | WeakHashingAlgorithms.cs:66:43:66:47 | access to local variable bytes | Insecure hashing algorithm (SHA1) depends on $@. | WeakHashingAlgorithms.cs:65:50:65:57 | access to parameter password | sensitive data (access to parameter password) |
